### PR TITLE
Custom leader-election configuration for KafkaChannel

### PR DESCRIPTION
**Pull request by travis-minke-sap**
The semi-recent automated update of dependencies #1058 pulled in changes in **knative.dev/pkg/leader-election/config.go** which are breaking the startup of the KafkaChannel Webhook.  Specifically the default config timings were relaxed to be less frequent...

```
func defaultConfig() *Config {
	return &Config{
		Buckets:       1,
		LeaseDuration: 60 * time.Second,
		RenewDeadline: 40 * time.Second,
		RetryPeriod:   10 * time.Second,
	}
}
```

The KafkaChannel Webhook deployment has a 20 second delay on the initial liveness probe...
```
        livenessProbe:
          <<: *probe
          initialDelaySeconds: 20
```

So the webhook is restarted continually before it can finish the leader election setup.

This is a problem when uninstalling and re-installing in an existing namespace as the leases are not removed.  If the namespace is removed this shouldn't be an issue.  For "optional" components like eventing-kakfa that live in the knative-eventing namespace, though it is common.  There is a large Slack thread on this outage in the eventing-kafka channel on Jan 26, 2022.

## Proposed Changes

The goal of this PR is a quick rollback/override to use the prior configuration values so that build/releases may proceed.  If in the future we want to undo this and find another solution we can always do so.

- Add a new **config/channel/config/leader-election-configmap.yaml** with prior configuration values.
- Symlinks from **config/channel/consolidated** and **config/channel/distributed** to include the new ConfigMap
- Change both consolidated and distributed controller Deployments to use the new ConfigMap instead of relying on the standard knative-eventing leader-election ConfigMap.
- Add the CONFIG_LEADERELECTION_NAME env var to the Webhook Deployment(s) referencing the new ConfigMap.

### CODE REVIEW

**Comment by travis-minke-sap** [comment]
Hold for consensus on approach ; )
/hold

File: config/channel/distributed/500-controller-deployment.yaml [hunk 1]
```diff
@@ -70,9 +70,9 @@ spec:
             cpu: 20m
             memory: 25Mi
         volumeMounts:
-          - name: config-kafka
-            mountPath: /etc/config-kafka
-      volumes:
         - name: config-kafka
```
Comment (lionelvillard) [inline config/channel/distributed/500-controller-deployment.yaml:73]: the indent is off (but still correct) thinking there is a change when there is not?

Comment (travis-minke-sap) [reply config/channel/distributed/500-controller-deployment.yaml:73]: ah... intellij.... sigh... - I'll correct it.   pretty sure it works either way but good to stay consistent

Comment (lionelvillard) [reply config/channel/distributed/500-controller-deployment.yaml:73]: nvm, it's consistent with the rest

**Comment by evankanderson** [comment]
@evankanderson -- unexpected fallout from the pkg changes

Given the above scenario, it seems reasonable / prudent to check on webhook liveness probes and testing in all repos.

Opened https://github.com/knative/pkg/issues/2422 to investigate

**Pull request merged**